builtin/providers/librato/resource_librato_metric.go

```diff
@@ -0,0 +1,432 @@
+						"color": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"display_max": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"display_min": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"display_units_long": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"display_units_short": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"display_stacked": &schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"created_by_ua": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"gap_detection": &schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"aggregate": &schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceLibratoMetricCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*librato.Client)
+
+	metric := new(librato.Metric)
+	if a, ok := d.GetOk("name"); ok {
+		metric.Name = librato.String(a.(string))
+	}
+	if a, ok := d.GetOk("display_name"); ok {
+		metric.DisplayName = librato.String(a.(string))
+	}
+	if a, ok := d.GetOk("type"); ok {
+		metric.Type = librato.String(a.(string))
+	}
+	if a, ok := d.GetOk("description"); ok {
+		metric.Description = librato.String(a.(string))
+	}
+	if a, ok := d.GetOk("period"); ok {
+		metric.Period = librato.Uint(a.(uint))
+	}
+	if a, ok := d.GetOk("composite"); ok {
+		metric.Composite = librato.String(a.(string))
+	}
+
+	if a, ok := d.GetOk("attributes"); ok {
+
+		attributeData := a.([]interface{})
+		if len(attributeData) > 1 {
+			return fmt.Errorf("Only one set of attributes per metric is supported")
+		}
+
+		if len(attributeData) == 1 && attributeData[0] == nil {
+			return fmt.Errorf("No attributes found in attributes block")
+		}
+
+		attributeDataMap := attributeData[0].(map[string]interface{})
+		attributes := new(librato.MetricAttributes)
+
+		if v, ok := attributeDataMap["color"].(string); ok && v != "" {
+			attributes.Color = librato.String(v)
+		}
+		if v, ok := attributeDataMap["display_max"].(string); ok && v != "" {
+			attributes.DisplayMax = librato.String(v)
+		}
+		if v, ok := attributeDataMap["display_min"].(string); ok && v != "" {
+			attributes.DisplayMin = librato.String(v)
+		}
+		if v, ok := attributeDataMap["display_units_long"].(string); ok && v != "" {
+			attributes.DisplayUnitsLong = *librato.String(v)
+		}
+		if v, ok := attributeDataMap["display_units_short"].(string); ok && v != "" {
+			attributes.DisplayUnitsShort = *librato.String(v)
+		}
+		if v, ok := attributeDataMap["created_by_ua"].(string); ok && v != "" {
+			attributes.CreatedByUA = *librato.String(v)
+		}
+		if v, ok := attributeDataMap["display_stacked"].(bool); ok {
```

Comment:
I added a default, as I think we want to enable `display_stacked` to be set by the user.